fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -17,25 +17,43 @@
 class ConditionalSelectionRate(ClassificationMoment):
     """Generic fairness moment for selection rates.
 
-    This serves as the base class for both :class:`DemographicParity`
-    and :class:`EqualizedOdds`. The two are distinguished by
-    the events they define, which in turn affect the
-    `index` field created by :meth:`load_data()`.
+    This serves as the base class for :class:`DemographicParity` :class:`EqualizedOdds`, and
+    others. All subclasses can be used as difference-based constraints or ratio-based constraints.
+    
+    Difference-based constraints are written with a difference bound :math:`\\epsilon`.
+    Ratio-based constraints have a similar :math:`\\epsilon`, referred to as ratio bound slack
+    below. Additionally, they require a ratio bound. Constraints are set up for each group
+    compared to the overall population (unless further events are specified, e.g., in Equalized
+    Odds). Constraint violation for difference-based constraints starts if the difference between
+    a group and the overall population with regard to a utility exceeds the difference bound. For
+    ratio-based constraints we consider the difference between the overall population's utility
```

Comment:
Putting in the two equations here might be useful. If you'd rather not do that, can you link to the relevant section of the user guide?

Reply:
I would, if it existed. That one will take some significant time.

Reply:
Could you at least put in the two equations, rather than describing them in the text?

Reply:
Miro said in the past that we shouldn’t add that level of detail in docstrings. I don’t think I can properly explain formulas in here either. I’ll write the doc pages later this week and can link them then.

Reply:
I personally find it useful when the PR adds a method and the user guide at the same time, and the docstring refers to the user guide for further explanation.

Reply:
I agree that that would be the best way to do these things in future :-)

Reply:
@adrinjalali agreed! I guess I hadn't thought the organization with student groups through. They don't necessarily have enough time to do all the tasks required for the open source repo due to how long semesters are and other commitments. For the future we should do this differently and have all changes at the same time as you suggest. Perhaps we can just add to their PRs until they're ready, as opposed to completing them and then adding follow-ups.

Reply:
> For the future we should do this differently and have all changes at the same time as you suggest. Perhaps we can just add to their PRs until they're ready, as opposed to completing them and then adding follow-ups.

Either way is valid IMO. I'd say the best way is the way you're comfortable with. I'm used to adding to people's PRs usually rather than follow up PRs, but if you're happy with openning an issue/submitting a followup PR, then that's all good.

Reply:
I have made some tweaks in `mitigation.rst` (within this PR), so there is now the description of the constraints in `UtilityParity`, so you can just point to there.

Reply:
linked to user guide